src/Security/Member.php

```diff
@@ -1691,11 +1693,15 @@ public function changePassword($password)
         $this->Password = $password;
         $valid = $this->validate();
 
+        $this->extend('onBeforeChangePassword', $password, $valid);
+
         if ($valid->isValid()) {
             $this->AutoLoginHash = null;
             $this->write();
         }
 
+        $this->extend('onAfterChangePassword', $password, clone $valid);
```

Comment:
It's a nice idea but unnecessary to clone.

Reply:
Was following your comment "and an onAfterChangePassword which can take the result but not modify it" - will remove